src/webgl/material.js

```diff
@@ -718,171 +725,493 @@ p5.prototype.createFilterShader = function (fragSrc) {
  * @example
  * <div modernizr='webgl'>
  * <code>
- * // Note: A "uniform" is a global variable within a shader program.
+ * let fillShader;
  *
- * // Create a string with the vertex shader program.
- * // The vertex shader is called for each vertex.
  * let vertSrc = `
  * precision highp float;
- * uniform mat4 uModelViewMatrix;
- * uniform mat4 uProjectionMatrix;
+ * attribute vec3 aPosition;
+ * void main() {
+ *   gl_Position = vec4(aPosition, 1.0);
+ * }
+ * `;
+ *
+ * let fragSrc = `
+ * precision highp float;
+ * uniform float uMouseX;
+ * void main() {
+ *   vec3 color = vec3(uMouseX, 0.0, 1.0); // Color based on mouse X
+ *   gl_FragColor = vec4(color, 1.0);
+ * }
+ * `;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
+ *   shader(fillShader);
+ *   noStroke();
+ *   describe('A square with color changing based on mouse X without lighting.');
+ * }
+ *
+ * function draw() {
+ *   fillShader.setUniform('uMouseX', map(mouseX, 0, width, 0, 1));
+ *   plane(100, 100);
+ * }
+ * </code>
+ * </div>
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let fillShader;
  *
+ * let vertSrc = `
+ * precision highp float;
  * attribute vec3 aPosition;
- * attribute vec2 aTexCoord;
- * varying vec2 vTexCoord;
+ * uniform mat4 uModelViewMatrix;
+ * uniform mat4 uProjectionMatrix;
+ * varying vec3 vPosition;
  *
  * void main() {
- *   vTexCoord = aTexCoord;
- *   vec4 positionVec4 = vec4(aPosition, 1.0);
- *   gl_Position = uProjectionMatrix * uModelViewMatrix * positionVec4;
+ *   vPosition = aPosition;
+ *   gl_Position = uProjectionMatrix * uModelViewMatrix * vec4(aPosition, 1.0);
  * }
  * `;
  *
- * // Create a string with the fragment shader program.
- * // The fragment shader is called for each pixel.
  * let fragSrc = `
  * precision highp float;
+ * uniform vec3 uLightDir;
+ * varying vec3 vPosition;
  *
  * void main() {
- *   // Set each pixel's RGBA value to yellow.
- *   gl_FragColor = vec4(1.0, 1.0, 0.0, 1.0);
+ *   vec3 lightDir = normalize(uLightDir);
+ *   float brightness = dot(lightDir, normalize(vPosition));
+ *   brightness = clamp(brightness, 0.4, 1.0);
+ *   vec3 color = vec3(0.3, 0.5, 1.0);
+ *   color = color * brightness * 3.0;
+ *   gl_FragColor = vec4(color, 1.0);
  * }
  * `;
  *
  * function setup() {
  *   createCanvas(100, 100, WEBGL);
+ *   fillShader = createShader(vertSrc, fragSrc);
```

Comment:
I think, we could include a shaderHooks example for fillShader, by removing one example.